fem/field_interpolant.hpp

```diff
@@ -0,0 +1,191 @@
+
+      const FiniteElement &el = *fes->GetFE(0);
+      const IntegrationRule *ir = &(IntRules.Get(el.GetGeomType(),
+                                                 2 * el.GetOrder() + 1));
+
+      L2->AddDomainIntegrator(new MassIntegrator(ir));
+      L2->Assemble();
+   }
+   // This function takes a vector quadrature function coefficient and projects it onto a GridFunction that lives
+   // in L2 space. This function requires fes to be L2 finite element space that we're projecting onto.
+   void ProjectQuadratureDiscCoefficient(GridFunction &gf,
+                                         VectorQuadratureFunctionCoefficient &vqfc,
+                                         FiniteElementSpace &fes);
+   // This function takes a quadrature function coefficient and projects it onto a GridFunction that lives
+   // in L2 space. This function requires fes to be L2 finite element space that we're projecting onto.
+   void ProjectQuadratureDiscCoefficient(GridFunction &gf,
+                                         QuadratureFunctionCoefficient &qfc,
+                                         FiniteElementSpace &fes);
+   // This function takes a vector quadrature function coefficient and projects it onto a GridFunction of the same space as vector
+   // quadrature function coefficient.
+   void ProjectQuadratureCoefficient(GridFunction &gf,
+                                     VectorQuadratureFunctionCoefficient &vqfc,
+                                     FiniteElementSpace &fes);
+   // This function takes a quadrature function coefficient and projects it onto a GridFunction of the same space as
+   // quadrature function coefficient.
+   void ProjectQuadratureCoefficient(GridFunction &gf,
+                                     QuadratureFunctionCoefficient &qfc,
+                                     FiniteElementSpace &fes);
+   // Tells the ProjectQuadratureDiscCoefficient that they need to recalculate the data.
+   void SetupDiscReset() { setup_disc = false; }
+   // Tells the ProjectQuadratureCoefficient that they need to recalculate the data.
+   virtual void FullReset()
+   {
+      L2->Update();
+      L2->Assemble();
+   }
+   virtual void SetupCG(double rel_tol = 1e-15, double abs_tol = 0.0,
+                        int print_level = 0, int max_iter = 2000)
+   {
+      cg = new CGSolver();
+      cg->SetPrintLevel(print_level);
+      cg->SetMaxIter(max_iter);
+      cg->SetRelTol(rel_tol);
+      cg->SetAbsTol(abs_tol);
+   }
+   ~FieldInterpolant()
+   {
+      delete L2;
+      delete cg;
+   }
+};
+
+#ifdef MFEM_USE_MPI
+class ParFieldInterpolant : public FieldInterpolant
+{
+protected:
+   ParBilinearForm *ParL2;
+public:
+   // The ParFiniteElementSpace passed into here should have a vdim set to 1 in order for the
+   // MassIntegrator to work properly if the ProjectQuadratureCoefficient method is used with
+   // a VectorQuadratureFunctionCoefficient.
+   ParFieldInterpolant(ParFiniteElementSpace *pfes) : FieldInterpolant(pfes)
```

Comment:
This calls the other constructor, which allocates its own data, which is probably not used.

Reply:
I created a protected default constructor on the parent class which doesn't allocate anything. I hope this is an acceptable way to solve this issue.